### DISCUSSION TITLE
Intercept with service worker (msw)

### DISCUSSION THREAD

**Discussion by hedleygois** [cypress-io/cypress#14390] [Questions and Help]
Hi,

I'm using https://mswjs.io/ to provide some data and cypress to test my application. The problem is that `cy.intercept` does not seem to fire. The requests are always hitting the service worker even if I use `forceNetworkError`, in fact, even if I use a function it won't fire. Is there a way to handle these scenarios where there's a SW?

Thanks!

**Comment by singhAmandeep007** [comment]
@hedleygois Check this out, might help

I have prepared a React + Vite boilerplate which includes integration of MSW with Vitest and Cypress.

This allows to easily test an application with mocked responses.
With mocked response you can easily simulate a success scenario as well as failure scenario, thus covering most of user stories.

What's included ?
✅ Automatic reset
✅ Isolation
✅ Logging
✅ Fully Type safe
✅ With middleware

[Repo](https://github.com/singhAmandeep007/react-vite-boilerplate)
[MSW + Cypress](https://github.com/singhAmandeep007/react-vite-boilerplate/blob/main/cypress/support/msw.ts)
[MSW + Vitest](https://github.com/singhAmandeep007/react-vite-boilerplate/blob/main/src/tests/testServer.ts)

Please give it a ⭐️ if it helped.
